tests/integration/ha/helpers.py

```diff
@@ -174,6 +186,103 @@ async def assert_continuous_writes_consistency(c_writes: ContinuousWrites) -> No
     assert result.max_stored_id == result.last_expected_id
 
 
+async def secondary_up_to_date(ops_test: OpsTest, unit_ip, expected_writes) -> bool:
+    """Checks if secondary is up to date with the cluster.
+
+    Retries over the period of one minute to give secondary adequate time to copy over data.
+    """
+    get_secondary_writes = await http_request(
+        ops_test,
+        "GET",
+        f"https://{unit_ip}:9200/series_index",
+    )
+    logger.error(get_secondary_writes)
+    assert get_secondary_writes == expected_writes
+
+
+def cut_network_from_unit(machine_name: str) -> None:
+    """Cut network from a lxc container.
+
+    Args:
+        machine_name: lxc container hostname
+    """
+    # apply a mask (device type `none`)
+    cut_network_command = f"lxc config device add {machine_name} eth0 none"
+    subprocess.check_call(cut_network_command.split())
+
+
+def restore_network_for_unit(machine_name: str) -> None:
+    """Restore network from a lxc container.
+
+    Args:
+        machine_name: lxc container hostname
+    """
+    # remove mask from eth0
+    restore_network_command = f"lxc config device remove {machine_name} eth0"
+    subprocess.check_call(restore_network_command.split())
+
+
+@retry(stop=stop_after_attempt(15), wait=wait_fixed(15))
+def wait_network_restore(model_name: str, hostname: str, old_ip: str) -> None:
+    """Wait until network is restored.
+
+    Args:
+        model_name: The name of the model
+        hostname: The name of the instance
+        old_ip: old registered IP address
+    """
+    if instance_ip(model_name, hostname) == old_ip:
+        raise Exception("Network not restored, IP address has not changed yet.")
+
+
+def instance_ip(model: str, instance: str) -> str:
```

Comment:
Could you use `get_application_unit_ids_ips(ops_test)[unit_id]`? it's broader than `unit_ip` as it gives us a full view and we get to stay consistent with the rest of the tests